src/main/java/org/terasology/taskSystem/TaskManagementSystem.java

```diff
@@ -344,4 +349,56 @@ public boolean assignAdvancedTaskToOreon(Actor oreon, String assignedTaskType) {
 
             return true;
     }
+
+    /**
+     * Receives the HorizontalCollisionEvent and decides whether the Oreon should abandon tasks
+     */
+    @ReceiveEvent(priority = EventPriority.PRIORITY_CRITICAL)
+    public void receiveCollisionEvent(HorizontalCollisionEvent collisionEvent, EntityRef oreon, MinionMoveComponent moveComponent) {
+        if (lastCollisionLocation == null) {
+            lastCollisionLocation = collisionEvent.getLocation();
+        } else {
+            if (lastCollisionLocation.getX() == collisionEvent.getLocation().getX() && lastCollisionLocation.getZ() == collisionEvent.getLocation().getZ()) {
```

Comment:
You are comparing floating-point values with '=='.   That may not work all of the time.
In this case, you are better off using as BaseVector3f.equals() seems to have some extra logic in there.
It's also a lot shorter and easier to read.

`
if (lastCollisionLocation..equals(collisionEvent.getLocation())) {
`

I guess the only question is whether checking the Y location matters.
If you do not want to consider the Y location, then maybe using the code that BaseVector3f.equals() uses would be a better approach.

`
            return Float.floatToIntBits(getX()) == Float.floatToIntBits(other.getX())
                && Float.floatToIntBits(getY()) == Float.floatToIntBits(other.getY())
                && Float.floatToIntBits(getZ()) == Float.floatToIntBits(other.getZ());
`